bin/run-all-the-things.php

```diff
@@ -38,5 +39,18 @@
         )
     );
 
+    Loop::addSignal(SIGINT, $func = function ($signal) use (&$func) {
+        echo 'Received signal: ', (string)$signal, PHP_EOL;
+
+        Loop::removeSignal(SIGINT, $func);
+        Loop::get()->stop();
+    });
+    Loop::addSignal(SIGTERM, $func = function ($signal) use (&$func) {
+        echo 'Received signal: ', (string)$signal, PHP_EOL;
+
+        Loop::removeSignal(SIGTERM, $func);
+        Loop::get()->stop();
+    });
+
```

Comment:
Either

```suggestion
    Loop::addSignal(SIGINT, $func = function ($signal) use (&$func) {
        echo 'Received signal: ', (string)$signal, PHP_EOL;

        Loop::get()->stop();
    });
    Loop::addSignal(SIGTERM, $func = function ($signal) use (&$func) {
        echo 'Received signal: ', (string)$signal, PHP_EOL;

        Loop::get()->stop();
    });

```

or

```suggestion
   $func = function ($signal) use ($func) {
        echo 'Received signal: ', (string)$signal, PHP_EOL;

        Loop::removeSignal(SIGINT, $func);
        Loop::removeSignal(SIGTERM, $func);
        Loop::get()->stop();
    };
    Loop::addSignal(SIGINT, $func);
    Loop::addSignal(SIGTERM, $func);

```

But probably also a stop function, if it doesn't exists on Ratchet, would help.